modules/twinklewarn.js

```diff
@@ -1144,20 +1149,21 @@ Twinkle.warn.callback.change_category = function twinklewarnCallbackChangeCatego
 		// no categories, just create the options right away
 		createEntries(Twinkle.warn.messages[value], sub_group, true);
 	} else if (value === 'singlecombined') {
-		createEntries(Twinkle.warn.messages['singlenotice'], sub_group, true);
-		createEntries(Twinkle.warn.messages['singlewarn'], sub_group, true);
+		var singletMessages = $.extend({}, Twinkle.warn.messages['singlenotice'], Twinkle.warn.messages['singlewarn']);
+		createEntries(singletMessages, sub_group, true);
 
 		// Sort the newly combined menu, preserving messageData
 		var selectGroup = $('select[name=sub_group]');
 		// .children() of level1, etc. will be populated with optgroup
 		// rather than option, but the singlet menus don't have groups
+		var selectedVal = selectGroup.val(); // Preserve selection after sorting
 		selectGroup.children('option').detach().sort(function(a, b) {
 			return a.text > b.text ? 1 : -1;
 		}).appendTo(selectGroup);
-		selectGroup[0].selectedIndex = 0; // Select the first item
+		selectGroup.val(selectedVal);
 	} else if (value === 'custom') {
 		createEntries(Twinkle.getPref('customWarningList'), sub_group, true);
-	} else {
+	} else if (value !== 'kitchensink') { // Levels 1-4im
```

Comment:
better to put the if (value === kitchensink) part first. Or use a switch instead of this if-else-if ladder.

Reply:
I think I just liked the "all of the above"-ness of it?  A switch does that just fine.